src/AnalysisPrograms/Recognizers/KoalaV3.cs

```diff
@@ -0,0 +1,144 @@
+// <copyright file="KoalaV3.cs" company="QutEcoacoustics">
+// All code in this file and all associated files are the copyright and property of the QUT Ecoacoustics Research Group (formerly MQUTeR, and formerly QUT Bioacoustics Research Group).
+// </copyright>
+
+namespace AnalysisPrograms.Recognizers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+    using System.Runtime.CompilerServices;
+    using Acoustics.Shared.ConfigFile;
+    using AnalysisBase;
+    using AnalysisPrograms.Recognizers.Base;
+    using AudioAnalysisTools;
+    using AudioAnalysisTools.DSP;
+    using AudioAnalysisTools.Events;
+    using AudioAnalysisTools.Events.Types;
+    using AudioAnalysisTools.Indices;
+    using AudioAnalysisTools.StandardSpectrograms;
+    using AudioAnalysisTools.WavTools;
+    using log4net;
+    using SixLabors.ImageSharp;
+    using TowseyLibrary;
+    using static AnalysisPrograms.Recognizers.GenericRecognizer;
+    using Path = System.IO.Path;
+
+    /// <summary>
+    /// A recognizer for the Australian Boobook Owl, /// https://en.wikipedia.org/wiki/Australian_boobook .
+    /// </summary>
+    internal class KoalaV3 : RecognizerBase
+    {
+        private static readonly ILog BoobookLog = LogManager.GetLogger(MethodBase.GetCurrentMethod().DeclaringType);
+
+        public override string Author => "Towsey";
+
+        public override string SpeciesName => "PhascolarctosCinereus";
+
+        public override string Description => "[ALPHA] Detects acoustic events for the Koala.";
+
+        public override AnalyzerConfig ParseConfig(FileInfo file)
+        {
+            RuntimeHelpers.RunClassConstructor(typeof(KoalaConfig).TypeHandle);
+            var config = ConfigFile.Deserialize<KoalaConfig>(file);
+
+            // validation of configs can be done here
+            GenericRecognizer.ValidateProfileTagsMatchAlgorithms(config.Profiles, file);
+
+            // This call sets a restriction so that only one generic algorithm is used.
+            // CHANGE this to accept multiple generic algorithms as required.
+            //if (result.Profiles.SingleOrDefault() is ForwardTrackParameters)
+            if (config.Profiles?.Count == 1 && config.Profiles.First().Value is ForwardTrackParameters)
+            {
+                return config;
+            }
+
```

Comment:
Is this recogniser in complete? I thought you wanted to detect harmonics as well?

Reply:
This is work in progress. I should have left a message to this effect at top of class.